server/src/main/java/org/opensearch/indices/replication/SegmentReplicator.java

```diff
@@ -278,7 +329,9 @@ public void onResponse(Void o) {
                 logger.debug(() -> new ParameterizedMessage("Finished replicating {} marking as done.", target.description()));
                 pruneCheckpointsUpToLastSync(target.indexShard());
                 onGoingReplications.markAsDone(replicationId);
-                if (target.state().getIndex().recoveredFileCount() != 0 && target.state().getIndex().recoveredBytes() != 0) {
+                if (target.state().getIndex().recoveredFileCount() != 0
+                    && target.state().getIndex().recoveredBytes() != 0
+                    && null != completedReplications) {
```

Comment:
would we need `completedReplications` for tracking ongoing merge replications for stats purposes later?

Reply:
Lets circle back at this when thinking from stats point. Probably lets create a META issue for this whole effort and add a stats item there. We can brainstorm about this later.